opensrp-reveal/src/main/java/org/smartregister/reveal/util/AlertDialogUtils.java

```diff
@@ -22,4 +24,22 @@ public static AlertDialog displayNotification(Context context, int title, int me
     public static AlertDialog displayNotification(Context context, String message) {
         return new AlertDialog.Builder(context).setMessage(message).setTitle(R.string.fetch_structures_title).setPositiveButton(R.string.ok, null).show();
     }
+
+    public static AlertDialog displayNotificationWithCallback(Context context, int title, int message, int positiveBtnTitle, int negativeBtnTitle, AlertDialogCallback callback, Object... formatArgs) {
+        AlertDialog alert = null;
+
+        if (formatArgs.length == 0)
+            alert = new AlertDialog.Builder(context).setMessage(message).setTitle(title).setPositiveButton(positiveBtnTitle, null).setNegativeButton(negativeBtnTitle,null).show();
+        else
+            alert = new AlertDialog.Builder(context).setMessage(context.getString(message, formatArgs)).setTitle(title).setPositiveButton(positiveBtnTitle, null).setNegativeButton(negativeBtnTitle,null).show();
+
+        alert.getButton(AlertDialog.BUTTON_POSITIVE).setOnClickListener(new View.OnClickListener() {
```

Comment:
Pass the listener when creating the alert Dialog instead of setting custom listener afterwards, Also use `DialogInterface.OnClickListener` instead of `View.OnClickListener`